core/src/main/java/org/apache/calcite/plan/RelOptUtil.java

```diff
@@ -1111,9 +1111,11 @@ private static void splitJoinCondition(
 
         boolean foundInput = false;
         for (int i = 0; i < inputs.size() && !foundInput; i++) {
-          final int lowerLimit = inputsRange[i].nextSetBit(0);
-          final int upperLimit = inputsRange[i].length();
-          if (projRefs.nextSetBit(lowerLimit) < upperLimit) {
+          final ImmutableBitSet inputRange = ImmutableBitSet.range(
+              inputsRange[i].nextSetBit(0),
+              inputsRange[i].length());
+
+          if (inputRange.contains(projRefs)) {
             leftInput = i;
```

Comment:
I think inputRange is a continuous range, so you don't need to create a new range:

```java
          if (inputsRange[i].contains(projRefs)) {
```

will suffice.